gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -666,13 +687,24 @@ public static class GobblinServiceJob extends BaseGobblinJob implements Interrup
 
     @Override
     public void executeImpl(JobExecutionContext context) throws JobExecutionException {
-      _log.info("Starting FlowSpec " + context.getJobDetail().getKey());
+      JobDetail jobDetail = context.getJobDetail();
+      _log.info("Starting FlowSpec " + jobDetail.getKey());
 
-      JobDataMap dataMap = context.getJobDetail().getJobDataMap();
+      JobDataMap dataMap = jobDetail.getJobDataMap();
       JobScheduler jobScheduler = (JobScheduler) dataMap.get(JOB_SCHEDULER_KEY);
       Properties jobProps = (Properties) dataMap.get(PROPERTIES_KEY);
       JobListener jobListener = (JobListener) dataMap.get(JOB_LISTENER_KEY);
 
+      // Obtain trigger timestamp from trigger to pass to jobProps
+      Trigger trigger = context.getTrigger();
+      // THIS current event has already fired if this method is called, so it now exists in <previousFireTime>
+      long triggerTimestampMillis = trigger.getPreviousFireTime().getTime();
+      jobProps.setProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_TRIGGER_TIMESTAMP_MILLIS_KEY,
+          String.valueOf(triggerTimestampMillis));
+      _log.info("Scheduler trigger tracing: [flowName: {} flowGroup: {}] - triggerTime: {} nextTriggerTime: {} - "
+              + "Job triggered by scheduler",
+          jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY), jobProps.getProperty(ConfigurationKeys.JOB_GROUP_KEY),
+          triggerTimestampMillis, trigger.getNextFireTime().getTime());
```

Comment:
somewhat surprised to see you work again from first principles rather than leverage the new method abstraction you just added as `scheduleJobLogUtil`...

but not a big deal (I see how the method would need to be more general for it to work here as well)